src/esp/gfx_batch/Renderer.cpp

```diff
@@ -1030,6 +1186,49 @@ void Renderer::draw(Mn::GL::AbstractFramebuffer& framebuffer) {
     scene.transformationUniform.setData(
         state_->absoluteTransformationsSorted.prefix(
             scene.transformationIdsSorted.size()));
+
+    /* Finish transformation-dependent per-draw info, upload it */
+    for (std::size_t i = 0; i != scene.transformationIds.size(); ++i) {
+      scene
+          .drawsSorted[i]
+          /* Extract normal matrix */
+          .setNormalMatrix(state_->absoluteTransformationsSorted[i]
+                               .transformationMatrix.normalMatrix())
+          // TODO light culling should happen here
+          .setLightOffsetCount(0, scene.lights.size());
```

Comment:
Where is maxLightCount enforced?

Reply:
On line 831, in `setLightCount()` passed to [Shaders::PhongGL::Configuration](https://doc.magnum.graphics/magnum/classMagnum_1_1Shaders_1_1PhongGL_1_1Configuration.html) -- it's a shader-compile-time setting, not a uniform input.

[As I said above](https://github.com/facebookresearch/habitat-sim/pull/1970#discussion_r1063616162), this will change slightly when light culling is done -- the compile-time light count will be handled internally, and the max per-draw light count will be limiting the count of lights used per draw, picking just the ones with most influence.